ocaml/xapi/xapi_sr.ml

```diff
@@ -240,12 +240,8 @@ let create  ~__context ~host ~device_config ~(physical_size:int64) ~name_label ~
   end;
   Helpers.call_api_functions ~__context
     (fun rpc session_id ->
-       List.iter
-         (fun self ->
-            try
-              Client.PBD.plug ~rpc ~session_id ~self
-            with e -> warn "Could not plug PBD '%s': %s" (Db.PBD.get_uuid ~__context ~self) (Printexc.to_string e))
-         pbds);
+       let tasks = List.map (fun self -> Client.Async.PBD.plug ~rpc ~session_id ~self) pbds in
+       Tasks.wait_for_all ~rpc ~session_id ~tasks);
```

Comment:
Note that `wait_for_all` does not fail if tasks have errored, so this should have retained the old partial success behaviour